tests/test_http_client.py

```diff
@@ -83,26 +73,27 @@ def test_credential():
     assert recorder[0].method == 'GET'
     assert recorder[0].url == _TEST_URL
     assert recorder[0].headers['Authorization'] == 'Bearer mock-token'
+
+def test_default_timeout():
+    client = _http_client.HttpClient()
+    assert client.timeout == _http_client.DEFAULT_TIMEOUT_SECONDS
+    recorder = _instrument(client, 'body')
+    client.request('get', _TEST_URL)
+    assert len(recorder) == 1
     assert recorder[0]._extra_kwargs['timeout'] == pytest.approx(
         _http_client.DEFAULT_TIMEOUT_SECONDS, 0.001)
 
-@pytest.mark.parametrize('timeout', [7, None])
-def test_timeout(timeout):
+@pytest.mark.parametrize('timeout', [7, 0, None])
+def test_custom_timeout(timeout):
```

Comment:
This function looks mostly identical to test_default_timeout. You could optionally refactor to something like this:

```python
def _helper_test_timeout(**kwargs):
    client = _http_client.HttpClient()
    timeout = kwargs.get('timeout', _http_client.DEFAULT_TIMEOUT_SECONDS)
    assert client.timeout == timeout
    ...
    if timeout is None
        assert recorder[0]._extra_kwargs['timeout'] is None
    else:
        assert ... == pytest.approx(timeout, 0.001)

def test_default_timeout():
    _helper_test_timeout()

@parameterize(...)
def test_custom_timeout(timeout)
    _helper_test_timeout(timeout=timeout)
```

(Or you could parameterize the helper directly.)

Reply:
Refactored with parameterization.